chia/full_node/full_node.py

```diff
@@ -1277,15 +1358,56 @@ async def add_block_batch(
         # Precondition: All blocks must be contiguous blocks, index i+1 must be the parent of index i
         # Returns a bool for success, as well as a StateChangeSummary if the peak was advanced
 
+        pre_validate_start = time.monotonic()
+        blockchain = AugmentedBlockchain(self.blockchain)
+        blocks_to_validate = await self.filter_blocks(blockchain, all_blocks, fork_info, vs)
+
+        if len(blocks_to_validate) == 0:
+            return True, None, None
+
+        futures = await self.prevalidate_blocks(
+            blockchain,
+            blocks_to_validate,
+            peer_info,
+            copy.copy(vs),
+            wp_summaries,
+        )
+        pre_validation_results = list(await asyncio.gather(*futures))
+
+        agg_state_change_summary, err = await self.add_prevalidated_blocks(
+            blockchain,
+            blocks_to_validate,
+            pre_validation_results,
+            fork_info,
+            peer_info,
+            vs,
+        )
+
+        if agg_state_change_summary is not None:
+            self._state_changed("new_peak")
+            self.log.debug(
+                f"Total time for {len(blocks_to_validate)} blocks: {time.monotonic() - pre_validate_start}, "
+                f"advanced: True"
+            )
+        return err is None, agg_state_change_summary, err
+
+    async def filter_blocks(
```

Comment:
i think this name is confusing since it suggest that the order is not relevant 
and this should only be relevant for the blocks under the actual fork point (not the one we are guessing from the wp) so after a few blocks this becomes irrelevant in the long sync and we can stop calling it on blocks

Reply:
right, it's (presumably) only relevant for the first batch. Do you have a suggestion for a better name?
I suppose I could also comment this (and maybe assert that this understanding holds up)

Reply:
skip_known_blocks or something i guess, maybe start_from_fork

Reply:
yes. `fork_info` is updated by this function

Reply:
yeah, I've been trying to think of ways to only do the skipping once too. The code to pop jobs from the queue would have to be duplicated in that case so it didn't seem like a clear-cut win. if we ever make an abstraction for an asyncio job pipeline, where all the queue management is handled in shared code, maybe it would make it more practical.